src/refresh-indicator.jsx

```diff
@@ -0,0 +1,264 @@
+
+    const [beginDeg, endDeg] = this._getArcDeg();
+    const arcLen = (endDeg - beginDeg) * perimeter / 360;
+    const dashOffset = -beginDeg * perimeter / 360;
+
+    const theme = this._getTheme();
+    return {
+      style: {
+        strokeDasharray: arcLen + ', ' + (perimeter - arcLen),
+        strokeDashoffset: dashOffset,
+        stroke: (isLoading || this.props.percentage === 100) ? theme.loadingStrokeColor : theme.strokeColor,
+        strokeLinecap: 'round',
+        opacity: p1,
+        strokeWidth: circle.strokeWidth * p1,
+        fill: 'none',
+      },
+      attr: {
+        cx: circle.originX,
+        cy: circle.originY,
+        r: circle.radiu,
+      },
+    };
+  },
+
+  _getPolygonStyle() {
+    const p1 = this._getFactor();
+    const circle = this._getCircleAttr();
+
+    const triangleCx = circle.originX + circle.radiu;
+    const triangleCy = circle.originY;
+    const dx = (circle.strokeWidth * 7 / 4) * p1;
+    const trianglePath = (triangleCx - dx) + ',' + triangleCy + ' ' + (triangleCx + dx) + ',' + triangleCy + ' ' + triangleCx + ',' + (triangleCy + dx);
+
+    const [, endDeg] = this._getArcDeg();
+
+    const theme = this._getTheme();
+    return {
+      style: {
+        fill: this.props.percentage === 100 ? theme.loadingStrokeColor : theme.strokeColor,
+        transform: `rotate(${endDeg}deg)`,
+        transformOrigin: `${circle.originX}px ${circle.originY}px`,
+        opacity: p1,
+      },
+      attr: {
+        points: trianglePath,
+      },
+    };
+  },
+
+  _scalePath(path, step) {
+    if (this.props.status !== 'loading') return;
+
+    const currStep = (step || 0) % 3;
+
+    clearTimeout(this._timer1);
+    this._timer1 = setTimeout(this._scalePath.bind(this, path, currStep + 1), currStep ? 750 : 250);
+
+    const circle = this._getCircleAttr();
+    const perimeter = Math.PI * 2 * circle.radiu;
+    const arcLen = perimeter * 0.64;
+
+    if (currStep === 0) {
+      path.style.strokeDasharray = '1, 200';
+      path.style.strokeDashoffset = 0;
+      path.style[this.prefixed('transitionDuration')] = '0ms';
+    } else if (currStep === 1) {
+      path.style.strokeDasharray = arcLen + ', 200';
+      path.style.strokeDashoffset = -15;
+      path.style[this.prefixed('transitionDuration')] = '750ms';
+    } else {
+      path.style.strokeDasharray = arcLen + ',200';
+      path.style.strokeDashoffset = -(perimeter - 1);
+      path.style[this.prefixed('transitionDuration')] = '850ms';
+    }
+  },
+
+  _rotateWrapper(wrapper) {
+    if (this.props.status !== 'loading') return;
+
+    clearTimeout(this._timer2);
+    this._timer2 = setTimeout(this._rotateWrapper.bind(this, wrapper), 10050);
+
+    AutoPrefix.set(wrapper.style, "transform", null);
+    AutoPrefix.set(wrapper.style, "transform", "rotate(0deg)");
+    AutoPrefix.set(wrapper.style, "transitionDuration", "0ms");
+
+    setTimeout(() => {
+      AutoPrefix.set(wrapper.style, "transform", "rotate(1800deg)");
```

Comment:
@oliviertassinari Thanks! I'll fix it. But the `isMounted()` method maybe deprecated in the future facebook/react#3220. However, I think it's ok to use it at this version.